src/analyzer/protocol/quic/QUIC.spicy

```diff
@@ -280,7 +280,7 @@ public type LongHeaderPacket = unit {
 };
 
 # A QUIC Frame.
-public type Frame = unit(header: LongHeaderPacket, from_client: bool, crypto_sink: sink) {
+public type Frame = unit(header: LongHeaderPacket, from_client: bool, inout crypto_sink: sink) {
```

Comment:
It seems that before passing an immutable value was fine since our refs implemented interior mutability (i.e., a const ref did seem to allow mutable access to the wrapped data) ... is that correct? If it is, I am not sure I saw something explicitly checking the previous or newer semantics explicitly in a test, but I think that would be great to have.

Separately I wonder whether it would make sense to more generally allow `copy` parameters in Spicy like we already do in HILTI. That would allow users to use narrower semantics (i.e., to specify that something is passed to be modified in a function body, but not require outside-visible changes). That might not be the semantics needed here, but I'd imagine it could be useful elsewhere.

Reply:
> It seems that before passing an immutable value was fine since our refs implemented interior mutability (i.e., a const ref did seem to allow mutable access to the wrapped data) ... is that correct?

That's correct I think (though not sure if it's the only case where currently we "miss" the constness). I'll see that I add a test on the Spicy side.

> Separately I wonder whether it would make sense to more generally allow copy parameters 

Yeah, I can see providing that.